### PR TITLE
Twitch: Expand app creation instructions

### LINKED ISSUE #31905: Twitch Documentation
### Feedback

when creating the Twitch OAuth -- make the name unique, if the "Save" does not "Save" then that means someone else on Twitch is using that name!

### URL

https://www.home-assistant.io/integrations/twitch/

### Version

2024.3.1

### Additional information

https://discuss.dev.twitch.com/t/i-cant-apply-for-twitch-api-integration/56767

### PR DESCRIPTION
**Pull request by jonnybergdahl**
## Proposed change

Added instructions for the Twitch integrations, as proposed by #31905.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [X] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #31905

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/twitch.markdown [hunk 1]
```diff
@@ -18,6 +18,10 @@ The Twitch integration will allow you to monitor [Twitch](https://www.twitch.tv/
 
 ## Get Twitch application credentials
 
-Create a new app at **Register Your Application** in the [Twitch developer portal](https://dev.twitch.tv/console/apps). Enter `https://my.home-assistant.io/redirect/oauth` in the **OAuth Redirect URL** field. Then, get the **Client ID** and **Client secret** from the new application.
+Create a new app at **Register Your Application** in the [Twitch developer portal](https://dev.twitch.tv/console/apps):
+
+ - Enter a **Name** for your app, note that it needs to be unique all over Twitch, so it is recommended to use your Twitch handle as part of the app name.
```
Comment (joostlek) [inline source/_integrations/twitch.markdown]: Why do we recommend this tho

**Comment by jonnybergdahl** [comment]
Because your Twitch handle works kinda like a namespace, it is unique all over Twitch.

**Comment by joostlek** [comment]
But i guess saying that something should be unique should be enough right? We dont care how they call it and we have no need for a special name, soooo

**Comment by jonnybergdahl** [comment]
And I don't care much for the sparse documentation you usually go for in the HA docs, often leaving newbies confused. Especially in this case where you get stuck creating the app name in Twitch without any indication that the reason is that it is not actually unique. I changed the wording. If you don't like it, I will close this PR.

File: source/_integrations/twitch.markdown [hunk 2]
```diff
@@ -18,6 +18,10 @@ The Twitch integration will allow you to monitor [Twitch](https://www.twitch.tv/
 
 ## Get Twitch application credentials
 
-Create a new app at **Register Your Application** in the [Twitch developer portal](https://dev.twitch.tv/console/apps). Enter `https://my.home-assistant.io/redirect/oauth` in the **OAuth Redirect URL** field. Then, get the **Client ID** and **Client secret** from the new application.
+Create a new app at **Register Your Application** in the [Twitch developer portal](https://dev.twitch.tv/console/apps):
+
+ - Enter a **Name** for your app. Note that it needs to be unique all over Twitch. You can consider incorporating your Twitch handle into the app name as one way to ensure uniqueness.
```
Comment (klaasnicolaas) [inline source/_integrations/twitch.markdown]:
```suggestion
- Enter a **Name** for your app. Note that it needs to be unique all over Twitch.
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]
Indicating that a name must be unique is sufficient, what users do with it is up to them.

**Comment by jonnybergdahl** [comment]
Compare the sparse wording on the Twitch integration with the verbose and helpful text of the Youtube integration. Documentation is supposed to be helpful for the user, and my changes add to that. 

With that said, I will not waste my time on this any more, neither will I accept your proposed change. You can merge or close this PR at your convenience, I don't care anymore.

**Review by klaasnicolaas — APPROVED** [review]

**Pull request merged**